src/tools/cephfs/DataScan.cc

```diff
@@ -844,6 +847,55 @@ int DataScan::scan_inodes()
   });
 }
 
+int DataScan::cleanup()
+{
+  bool roots_present;
+  int r = driver->check_roots(&roots_present);
+  if (r != 0) {
+    derr << "Unexpected error checking roots: '"
+      << cpp_strerror(r) << "'" << dendl;
+    return r;
+  }
+
+  if (!roots_present) {
+    std::cerr << "Some or all system inodes are absent.  Run 'init' from "
+      "one node before running 'cleanup'" << std::endl;
+    return -EIO;
+  }
+
+  return forall_objects(data_io, true, [this](
+        std::string const &oid,
+        uint64_t obj_name_ino,
+        uint64_t obj_name_offset) -> int
+      {
+      int r = 0;
+
+      AccumulateResult accum_res;
+      inode_backtrace_t backtrace;
+      file_layout_t loaded_layout = file_layout_t::get_default();
+      r = ClsCephFSClient::fetch_inode_accumulate_result(
+        data_io, oid, &backtrace, &loaded_layout, &accum_res);
+
+      if (r < 0) {
+      dout(4) << "No accumulated metadata found for '"
+      << oid << "': " << cpp_strerror(r) << dendl;
+      return r;
+      }
+
+      dout(4) << "Start cleanup accumulated metadata" << dendl;
+
+      r = ClsCephFSClient::delete_inode_accumulate_result(data_io, oid);
+
+      if (r < 0) {
+        dout(4) << "Error deleting accumulated metadata from '"
+          << oid << "': " << cpp_strerror(r) << dendl;
+        return r;
```

Comment:
In the rare event of an error here, we probably don't want to return -- for example if there is one object that can't be written, the user still probably wants to delete the attrs on all the other objects.